spec/presenters/content_type_resolver_spec.rb

```diff
@@ -0,0 +1,97 @@
+require "rails_helper"
+
+RSpec.describe Presenters::ContentTypeResolver do
+  subject { described_class.new("html") }
+
+  it "inlines content of the specified content type" do
```

Comment:
I designed it such that the system fails early on receiving content that isn't "well formed" through a custom validator. The scenario you describe is tested here:

https://github.com/alphagov/publishing-api/blob/forward-html-content-type-only/spec/validators/well_formed_content_types_validator_spec.rb#L67-L76

My thinking was that I'd fail early on receiving illy formed content and relieve the ContentTypeResolver of this responsibility. I'm happy to double test this though – but I'd be a little concerned that people would think it's acceptable to send illy formed content to the resolver.

What do you think?